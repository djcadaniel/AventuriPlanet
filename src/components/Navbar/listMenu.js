export const listMenu = [
  {nameRuta : 'Inicio', path : '/'},
  {nameRuta : 'Clases', path : '/clases'},
  {nameRuta : 'Especialidades', path : '/especialidades'},
  {nameRuta : 'Nosotros', path : '/nosotros'},
  {nameRuta : 'Contactanos', path : '/contactanos'},
]