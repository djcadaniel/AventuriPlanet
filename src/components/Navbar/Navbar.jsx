import React, { useState } from 'react';
import {BiMenu} from 'react-icons/bi';
import {IoMdClose} from 'react-icons/io';
import { Link } from 'react-router-dom';
import { listMenu } from './listMenu';

export const Navbar = () => {

  const [menu, setMenu] = useState(true);

  const onMenu = ()=>{
    setMenu(!menu)
    console.log(menu);
  }
  const onMenu2 =()=>{
    if(!menu){
      setMenu(!menu)
    }
  }

  return (
    <header className=' relative bg-primary h-24 flex z-50'>
      <nav className='content  md:h-auto '>
        <div className='fixed top-0 left-0 right-0 flex flex-row justify-between items-center h-24 bg-primary px-5'>
          <div className='text-text-blanco'>Logo</div>
          <button 
            className='text-2xl md:hidden transition-all ease-in-out'
            onClick={onMenu}
          >
            {menu ? <div className='text-white'><BiMenu /></div> : <div className='text-white'><IoMdClose /></div> }
          </button>
          <div 
            className={`${menu && "opacity-0"} md:hidden fixed inset-0 top-24 bg-gray-600/50 backdrop-blur-sm h-[calc(100vh-96px)]`}
            onClick={(menu)=>onMenu2(menu)}
          >
          </div>
          <ul className={`${menu && '-left-full md:left-0'} absolute md:relative inset-0 top-24 md:top-0 w-3/5 md:w-auto h-[calc(100vh-96px)] md:h-auto flex flex-col items-center justify-center md:flex-row transition-all ease-in-out text-text-blanco bg-primary`}>
            {
              listMenu.map((item, index) => (
                <div key={index}>
                  <Link 
                    to = {item.path}
                    className='mb-5 md:ml-5'
                    onClick={onMenu}
                  >
                    {item.nameRuta}
                  </Link>
                </div>
              ))
            }
          </ul>
        </div>
      </nav>
    </ header>
  )
}
