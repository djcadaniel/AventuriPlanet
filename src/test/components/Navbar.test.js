import {render} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import {Navbar} from '../../components/Navbar/Navbar';

describe('Pruebas en <NavBar />', () => {
  test('Debe de hacer match con el snapshot', () => {
    // console.log(render(<Navbar />));
    const {container} = render(<MemoryRouter><Navbar /></MemoryRouter>)
    expect(container).toMatchSnapshot();
  });
});