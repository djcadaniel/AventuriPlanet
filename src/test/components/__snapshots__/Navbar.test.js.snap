// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavBar /> Debe de hacer match con el snapshot 1`] = `
<div>
  <header
    class="relative bg-green-300 h-24 flex z-50"
  >
    <nav
      class="content flex flex-row md:h-auto justify-between items-center"
    >
      <div>
        Logo
      </div>
      <button
        class="text-2xl md:hidden transition-all ease-in-out"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
          />
        </svg>
      </button>
      <div
        class="opacity-0 md:hidden fixed inset-0 top-24 bg-gray-600/50 backdrop-blur-sm h-[calc(100vh-96px)]"
      />
      <ul
        class="-left-full md:left-0 absolute md:relative inset-0 top-24 md:top-0 w-3/5 md:w-auto h-[calc(100vh-96px)] md:h-auto flex flex-col md:flex-row items-center justify-center bg-red-300 transition-all ease-in-out"
      >
        <a
          class="mb-5 md:ml-5"
          href="/"
        >
          Inicio
        </a>
        <a
          class="mb-5 md:ml-5"
          href="/clases"
        >
          Clases
        </a>
        <a
          class="mb-5 md:ml-5"
          href="/especialidades"
        >
          Especialidades
        </a>
        <a
          class="mb-5 md:ml-5"
          href="/nosotros"
        >
          Nosotros
        </a>
        <a
          class="mb-5 md:ml-5"
          href="/contactanos"
        >
          Contactanos
        </a>
      </ul>
    </nav>
  </header>
</div>
`;
